response_operations_ui/common/credentials.py

```diff
@@ -0,0 +1,16 @@
+import requests
+from flask import current_app
+
+from response_operations_ui.oidc.oidc import OIDCCredentialsService
+
+
+def fetch_and_apply_oidc_credentials(session: requests.Session, client_id: str) -> None:
+    """
+    Raises:
+       GoogleAuthError: If there is an error fetching or applying OIDC credentials.
+    """
+    # Type ignore: oidc_credentials_service is a singleton of this application
+    oidc_credentials_service: OIDCCredentialsService = current_app.oidc["oidc_credentials_service"]  # type: ignore
+
+    credentials = oidc_credentials_service.get_credentials(iap_client_id=client_id)
+    credentials.apply(headers=session.headers)
```

Comment:
This is where the Google library magic happens that we don't need to care too much about. Integrates with their OIDC service layer and 'knows' the service account the app is running under and gets the token to add as the `bearer:` in the headers.